### PR TITLE
flatten-array: tests are now generated

### PR DESCRIPTION
**Pull request by ajwann**
## What?
Built the generator for flatten-array

## How Has This Been Tested?
```ruby
bin/generate
rake test
```

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Enhancement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or enhancement that would cause existing functionality to change)
- [ ] Gardening (code and/or documentation cleanup)

## References and Closures
references #396 

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [ ] My change relies on a pending issue/pull request
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: exercises/flatten-array/flatten_array_test.rb [hunk 1]
```diff
@@ -1,38 +1,63 @@
...
     skip
     fa = FlattenArray.flatten([0, 2, [[2, 3], 8, [[100]], nil, [[nil]]], -2])
     assert_equal [0, 2, 2, 3, 8, 100, -2], fa
   end
 
-  def test_all_values_are_nil
+  def test_all_values_in_nested_list_are_null
     skip
-    fa = FlattenArray.flatten([nil, [[[nil]]], nil, [[nil, nil], nil], nil])
+    fa = FlattenArray.flatten([nil, [[[nil]]], nil, nil, [[nil, nil], nil], nil])
```
Comment (ajwann) [inline exercises/flatten-array/flatten_array_test.rb]: It looks like the canonical data had an extra nil?

File: exercises/flatten-array/flatten_array_test.rb [hunk 2]
```diff
@@ -1,38 +1,63 @@
 require 'minitest/autorun'
 require_relative 'flatten_array'
 
+# Common test data version: 1.1.0 d7a3c09
 class FlattenArrayTest < Minitest::Test
-  def test_two_level_nesting
+  def test_no_nesting
+    # skip
+    fa = FlattenArray.flatten([0, 1, 2])
```
Comment (Insti) [inline exercises/flatten-array/flatten_array_test.rb]: > `fa`


Use a real word or phrase here please.

File: exercises/flatten-array/flatten_array_test.rb (hunk 1 above)
Comment (Insti) [reply exercises/flatten-array/flatten_array_test.rb]: I guess. It doesn't matter right?

File: exercises/flatten-array/.meta/solutions/flatten_array.rb [hunk 3]
```diff
@@ -1,6 +1,9 @@
-class FlattenArray
+module BookKeeping
   VERSION = 1
+end
+
+class FlattenArray
   def self.flatten(arr)
     arr.flatten.compact
```
Comment (Insti) [inline exercises/flatten-array/.meta/solutions/flatten_array.rb]: Good old Ruby.

**Comment by Insti** [comment]
Another good PR. Looks good @ajwann, just the `fa` to fix.

Although, if you want to argue that you're just converting what is already there I'd have to concede.
But while we're here we may as well fix it?

**Comment by Insti** [comment]
The `fa` thing wasn't really a big deal so I've merged this anyway.

Thanks again @ajwann :heart:

**Pull request merged**